lumigator/backend/backend/config_templates.py

```diff
@@ -151,28 +151,38 @@
 }}"""
 
 
-templates = {
-    JobType.INFERENCE: {
-        "default": default_infer_template,
-        "oai://gpt-4o-mini": oai_infer_template,
-        "oai://gpt-4o": oai_eval_template,
-        "mistral://open-mistral-7b": oai_infer_template,
-        "llamafile://mistralai/Mistral-7B-Instruct-v0.2": oai_infer_template,
-    },
-    JobType.EVALUATION: {
-        "default": causal_eval_template,
-        "hf://facebook/bart-large-cnn": bart_eval_template,
-        "hf://Falconsai/text_summarization": seq2seq_eval_template,
-        "hf://mistralai/Mistral-7B-Instruct-v0.3": causal_eval_template,
-        "oai://gpt-4o-mini": oai_eval_template,
-        "oai://gpt-4o": oai_eval_template,
-        "mistral://open-mistral-7b": oai_eval_template,
-        "llamafile://mistralai/Mistral-7B-Instruct-v0.2": oai_eval_template,
-    },
+def lookup_template(job_type, model_name):
+    if job_type == JobType.INFERENCE:
+        # Locally hosted models using oai client
+        oai_compatible_local_model_prefixes = ("ollama://", "llamafile://")
+        if model_name.startswith(oai_compatible_local_model_prefixes):
+            return oai_infer_template
+        inference_templates = {
+            "default": default_infer_template,
+            "oai://gpt-4o-mini": oai_infer_template,
+            "oai://gpt-4o": oai_infer_template,
+            "mistral://open-mistral-7b": oai_infer_template,
+        }
+        # If no config template is provided, use the default one for the job type
+        return inference_templates.get(model_name, default_infer_template)
+
+    elif job_type == JobType.EVALUATION:
```

Comment:
This could be just an `if` since there is a `return` statement in every block. Same for the next `elif` statement.